collab-service/src/controller/question_handler.ts

```diff
@@ -0,0 +1,43 @@
+import { ChannelCredentials } from '@grpc/grpc-js';
+import { QuestionServiceClient } from '../proto/question-service.grpc-client';
+import { Question, QuestionDifficulty } from '../proto/types';
+import loadEnvironment from '../utils/env_loader';
+
+const envConfig = loadEnvironment();
+
+const questionClient = new QuestionServiceClient(
+  envConfig.QUESTION_SERVER_URL,
+  ChannelCredentials.createInsecure(),
+  {},
+  {},
+);
+
+function getQuestionByDifficulty(difficulty: QuestionDifficulty): Promise<Question | undefined> {
```

Comment:
Controllers are more meant for handling upstream (requests), rather than downstream requests. You should rename this to be adapter instead to indicate that it is handling something downstream.